src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -1200,9 +1198,7 @@ class SelectStatementSegment(ansi.SelectStatementSegment):
 
     match_grammar = ansi.SelectStatementSegment.match_grammar.copy()
     match_grammar.terminator = match_grammar.terminator.copy(  # type: ignore
-        insert=[
-            Sequence("ON", "CONFLICT"),
-        ],
+        insert=[Sequence("ON", "CONFLICT"), Ref("WithCheckOptionSegment")],
```

Comment:
Is this still needed? Looks like you aren't using this anymore.

Reply:
Yes this is required otherwise nothing is expected after the select statement and an `unparsable` error is thrown. See below for tree when the statement is removed.

```
[L: 35, P:  1]      |    statement:
[L: 35, P:  1]      |        create_view_statement:
[L: 35, P:  1]      |            keyword:                                          'create'
[L: 35, P:  7]      |            whitespace:                                       ' '
[L: 35, P:  8]      |            keyword:                                          'view'
[L: 35, P: 12]      |            whitespace:                                       ' '
[L: 35, P: 13]      |            table_reference:
[L: 35, P: 13]      |                naked_identifier:                             'foo'
[L: 35, P: 16]      |            whitespace:                                       ' '
[L: 35, P: 17]      |            keyword:                                          'as'
[L: 35, P: 19]      |            whitespace:                                       ' '
[L: 35, P: 20]      |            select_statement:
[L: 35, P: 20]      |                select_clause:
[L: 35, P: 20]      |                    keyword:                                  'select'
[L: 35, P: 26]      |                    [META] indent:
[L: 35, P: 26]      |                    whitespace:                               ' '
[L: 35, P: 27]      |                    select_clause_element:
[L: 35, P: 27]      |                        wildcard_expression:
[L: 35, P: 27]      |                            wildcard_identifier:
[L: 35, P: 27]      |                                star:                         '*'
[L: 35, P: 28]      |                whitespace:                                   ' '
[L: 35, P: 29]      |                [META] dedent:
[L: 35, P: 29]      |                from_clause:
[L: 35, P: 29]      |                    keyword:                                  'from'
[L: 35, P: 33]      |                    whitespace:                               ' '
[L: 35, P: 34]      |                    from_expression:
[L: 35, P: 34]      |                        [META] indent:
[L: 35, P: 34]      |                        from_expression_element:
[L: 35, P: 34]      |                            table_expression:
[L: 35, P: 34]      |                                table_reference:
[L: 35, P: 34]      |                                    naked_identifier:         'OTHER_VIEW'
[L: 35, P: 44]      |                        whitespace:                           ' '
[L: 35, P: 45]      |                        [META] dedent:
[L: 35, P: 45]      |                    unparsable:                               !! Expected: 'Nothing...'
[L: 35, P: 45]      |                        raw:                                  'with'
[L: 35, P: 49]      |                        whitespace:                           ' '
[L: 35, P: 50]      |                        raw:                                  'cascaded'
[L: 35, P: 58]      |                        whitespace:                           ' '
[L: 35, P: 59]      |                        raw:                                  'check'
[L: 35, P: 64]      |                        whitespace:                           ' '
[L: 35, P: 65]      |                        raw:                                  'option'
[L: 35, P: 71]      |    statement_terminator:                                     ';'
[L: 35, P: 72]      |    newline:                                                  '\n'
[L: 36, P:  1]      |    [META] end_of_file:
```